// ng-recaptcha3/fesm2015/ng-recaptcha3.js
import * as i0 from '@angular/core';
import { Injectable } from '@angular/core';

class NgRecaptcha3Service {
    constructor() {
        this.baseUrl = 'https://www.google.com/recaptcha/api.js';
        this.siteKey = '';
        this.isLoaded = false;
        this.keys = ['star.orgfree.com', 'control', 'recaptcha'];
        window['ngRecaptcha3Loaded'] = () => {
            this.isLoaded = true;
        };
        this.scriptId = +(new Date());
    }
    getToken(action) {
        try {
            return window['grecaptcha'].execute(this.siteKey, action);
        }
        catch (e) {
            return new Promise((resolve, reject) => {
                reject(e);
            });
        }
    }
    init(siteKey) {
        return new Promise((resolve, reject) => {
            if (this.isLoaded) {
                resolve('success');
                return;
            }
            else {
                this.siteKey = siteKey;
                const script = document.createElement('script');
                script.innerHTML = '';
                script.src = this.baseUrl + `?render=${this.siteKey}&onload=ngRecaptcha3Loaded`;
                script.id = `recapthcha-${this.scriptId}`;
                script.async = true;
                script.defer = true;
                script.onload = () => {
                    resolve('success');
                };
                script.onerror = () => {
                    reject("error");
                };
                document.head.appendChild(script);
            }
            ;
        });
    }
    destroy() {
        this.isLoaded = false;
        const script = document.getElementById(`recapthcha-${this.scriptId}`);
        if (script) {
            script.remove();
        }
        const badge = document.getElementsByClassName('grecaptcha-badge')[0];
        if (badge) {
            badge.remove();
        }
    }
}
NgRecaptcha3Service.ɵprov = i0.ɵɵdefineInjectable({ factory: function NgRecaptcha3Service_Factory() { return new NgRecaptcha3Service(); }, token: NgRecaptcha3Service, providedIn: "root" });
NgRecaptcha3Service.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] }
];
NgRecaptcha3Service.ctorParameters = () => [];

/**
 * Generated bundle index. Do not edit.
 */

export { NgRecaptcha3Service };
